dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java

```diff
@@ -422,7 +422,6 @@ private void addProcessStopEvent(ProcessInstance processInstance) {
     private void addTaskRetryEvent(TaskInstance taskInstance) {
         TaskStateEvent stateEvent = TaskStateEvent.builder()
                 .processInstanceId(taskInstance.getProcessInstanceId())
-                .taskInstanceId(taskInstance.getId())
```

Comment:
Just curious, why you choose to avoid using `getId` here instead of generating and setting an id for the retried task instance?

Reply:
We only use the task instance Id to print logs in the TaskRetryStateEventHandler.
If an existing task instance id is set when the task is retried, no new task instance will be created

This is pr solved by setting the id: #13074, and I closed it because I think the new one is simpler to fix the bug.

WDYT?